sdk/communication/chat/src/main/java/com/azure/android/communication/chat/AzureCommunicationChatServiceAsyncClient.java

```diff
@@ -0,0 +1,490 @@
+     * @throws ErrorException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     */
+    public void deleteChatThread(String chatThreadId, final Callback<Void> callback) {
+        this.serviceClient.deleteChatThread(chatThreadId, callback);
+    }
+
+    /**
+     * Get the next page of items.
+     * 
+     * @param nextLink null
+     * @param callback the Callback that receives the response.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ErrorException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     */
+    public void listChatReadReceiptsNext(String nextLink, final Callback<Page<ReadReceipt>> callback) {
+        this.serviceClient.listChatReadReceiptsNext(nextLink, callback);
+    }
+
+    /**
+     * Get the next page of items.
+     * 
+     * @param nextLink null
+     * @param callback the Callback that receives the response.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ErrorException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     */
+    public void listChatMessagesNext(String nextLink, final Callback<Page<ChatMessage>> callback) {
+        this.serviceClient.listChatMessagesNext(nextLink, callback);
+    }
+
+    /**
+     * Get the next page of items.
+     * 
+     * @param nextLink null
+     * @param callback the Callback that receives the response.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ErrorException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     */
+    public void listChatThreadMembersNext(String nextLink, final Callback<Page<ChatThreadMember>> callback) {
+        this.serviceClient.listChatThreadMembersNext(nextLink, callback);
+    }
+
+    /**
+     * Get the next page of items.
+     * 
+     * @param nextLink null
+     * @param callback the Callback that receives the response.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ErrorException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     */
+    public void listChatThreadsNext(String nextLink, final Callback<Page<ChatThreadInfo>> callback) {
+        this.serviceClient.listChatThreadsNext(nextLink, callback);
+    }
+
+    /**
+     * A builder for creating a new instance of the AzureCommunicationChatServiceAsyncClient type.
+     */
+    public static final class Builder {
```

Comment:
Would we want to allow for users to provide other types of interceptors or would that be achieved by them adding them to the `serviceClientBuilder` and then passing that object to this builder?

Reply:
that's correct, for any additional interceptor customization, the user should use the core service-client.